OpenShift Compliance Operator 1.4.0

GroceryBoyJr opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.11+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/CMP-2315
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: 
This is the release note:
https://file.rdu.redhat.com/jbrigman/compliance-operator-140/security/compliance_operator/compliance-operator-release-notes.html

This is the change to the documentation that removes Procedure Step 1 and edits Procedure Step 2:
https://docs.openshift.com/container-platform/4.14/security/compliance_operator/co-scans/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

Vincent056 commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+[id="compliance-operator-1-4-0-new-features-and-enhancements"]
+=== New features and enhancements
+
+* Users can now pause scan schedules by setting the `ScanSetting.suspend` attribute to `True`. This allows users to suspend a scan and reactivate it without the need to delete and re-create the `ScanSettingBinding`. This simplifies pausing scans during maintenance periods. link:https://issues.redhat.com/browse/CMP-2123[CMP-2123]
```
should we mention here that we are pausing the scan schedule instead of the actual scan?

GroceryBoyJr replied:
@Vincent056 I see your point, thank you for that feedback. I'll change "suspend a scan" to "suspend a scan schedule"

Vincent056 commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -165,7 +196,7 @@ The following advisory is available for the OpenShift Compliance Operator 0.1.61
 * Before this update, the instructions for `ocp4-kubelet-configure-tls-cipher-suites` were incomplete, requiring users to refine the query manually. With this update, the query provided in `ocp4-kubelet-configure-tls-cipher-suites` returns the actual results to perform the audit steps. (link:https://issues.redhat.com/browse/OCPBUGS-3017[*OCPBUGS-3017*])
 
-* Before this update,`ScanSettingBinding` objects created without a `settingRef` variable did not use an appropriate default value. With this update, the `ScanSettingBinding` objects without a `settingRef` variable use the `default` value. (link:https://issues.redhat.com/browse/OCPBUGS-3420[*OCPBUGS-3420*])
+* Before this update,`ScanSettingBinding` objects created without a `settingRef` variable did not use an appropriate default value. With this update, the `ScanSettingBinding` objects without a `settingRef` variable use the `default` value.
```
I wonder if there is a reason why those link ref are being removed

sheriff-rh replied:
Hi Vincent - the links are removed because they are internal bugs. This is part of our new tooling update to catch references that customers cannot access or are not publicly available. 

@GroceryBoyJr  we need to remove the entire entry, including the description of the bug.

GroceryBoyJr replied:
This is done....thank you for the feedback!

BhargaviGudi commented:
@GroceryBoyJr Below bugs details are missing under bug fix
[OCPBUGS-11226](https://issues.redhat.com/browse/OCPBUGS-11226)	compliance operator: Improve the method to scan custom node pools via variable
[OCPBUGS-11481](https://issues.redhat.com/browse/OCPBUGS-11481)	OCP CIS Compliance version 0.1.61 issue
[OCPBUGS-11932](https://issues.redhat.com/browse/OCPBUGS-11932)	Compliance Operator 0.1.61 flagging OVS permissions on NSX cluster
[OCPBUGS-17494](https://issues.redhat.com/browse/OCPBUGS-17494)	Compliance Operator rprivate default mount propagation in combination with `hostPath: path: /` volume mounts of pods created breaks CSI driver relying on multipath
[OCPBUGS-3009](https://issues.redhat.com/browse/OCPBUGS-3009)	Runtime modification in the TailoredProfiles are not detected by the compliance operator and rules are not disabled
[OCPBUGS-8239](https://issues.redhat.com/browse/OCPBUGS-8239)	Compliance Operator rule: ocp4-kubelet-enable-streaming-connections is failing
[OCPBUGS-24594](https://issues.redhat.com/browse/OCPBUGS-24594)	Rule rhcos4-audit-rules-login-events-faillock will fail even after auto-remediation applied
[OCPBUGS-23341](https://issues.redhat.com/browse/OCPBUGS-23341)	The images in the profileparser containers and pauser of the profile parser pods are not updated when upgrade Compliance Operator from v1.1.0 to v1.20

GroceryBoyJr commented:
@BhargaviGudi Thank you for that feedback. I will add all bugs that are PUBLIC facing, I can no longer add bugs that are PRIVATE and not visible to customers.

GroceryBoyJr commented:
@BhargaviGudi I cannot add the following because they show as PRIVATE in the errata: 
[OCPBUGS-11226](https://issues.redhat.com/browse/OCPBUGS-11226)	compliance operator: Improve the method to scan custom node pools via variable
[OCPBUGS-11481](https://issues.redhat.com/browse/OCPBUGS-11481)	OCP CIS Compliance version 0.1.61 issue
[OCPBUGS-11932](https://issues.redhat.com/browse/OCPBUGS-11932)	Compliance Operator 0.1.61 flagging OVS permissions on NSX cluster
[OCPBUGS-3009](https://issues.redhat.com/browse/OCPBUGS-3009)	Runtime modification in the TailoredProfiles are not detected by the compliance operator and rules are not disabled
[OCPBUGS-8239](https://issues.redhat.com/browse/OCPBUGS-8239)	Compliance Operator rule: ocp4-kubelet-enable-streaming-connections is failing
[OCPBUGS-23341](https://issues.redhat.com/browse/OCPBUGS-23341)	The images in the profileparser containers and pauser of the profile parser pods are not updated when upgrade Compliance Operator from v1.1.0 to v1.20

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,39 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Previously, Compliance Operator remediations would be shown as "unapplied" after scans for compliance were completed. link:https://issues.redhat.com/browse/OCPBUGS-8041[OCPBUGS-8041]
+
+* Previously, `PodSecurity` alerts were generated by Compliance Operator after a cluster upgrade to 4.11. link:https://issues.redhat.com/brows/OCPBUGS-8071[OCPBUGS-8071]
```
This bug has now been dropped from the errata.

GroceryBoyJr replied:
@sheriff-rh I gotta keep my eyes on the prize errata!

GroceryBoyJr commented:
/label peer-review-needed

GroceryBoyJr commented:
/retest

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Compliance Operator now supports profile names in `ComplianceRules`. link:https://issues.redhat.com/browse/CMP-2126[CMP-2126]
+
+* Compliance Operator compatibility with improved cronjob API improvements is available in this release. link:https://issues.redhat.com/browse/CMP-2310[CMP-2310]
```
`cronjob` should have backticks.

GroceryBoyJr replied:
Fixed, thank you!

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+[id="compliance-operator-1-4-0-bug-fixes"]
+=== Bug fixes
+
+* With this update, rprivate default mount propagation is now handled correctly for root volume mounts of pods that rely on multipathing. link:https://issues.redhat.com/browse/OCPBUGS-17494[OCPBUGS-17494]
```
`rprivate` should have backticks.

GroceryBoyJr replied:
Done, thank you!

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* With this update, rprivate default mount propagation is now handled correctly for root volume mounts of pods that rely on multipathing. link:https://issues.redhat.com/browse/OCPBUGS-17494[OCPBUGS-17494]
+
+* RHCOS audit rule failure locks are now applied correctly after auto-remediation. link:https://issues.redhat.com/browse/OCPBUGS-24594[OCPBUGS-24594]
```
`RHCOS` can be an attribute:
{op-system-first} Red Hat Enterprise Linux CoreOS (RHCOS)
{op-system} RHCOS

GroceryBoyJr replied:
Done, thank you!

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* RHCOS audit rule failure locks are now applied correctly after auto-remediation. link:https://issues.redhat.com/browse/OCPBUGS-24594[OCPBUGS-24594]
+
+* This update repairs api-checks-pod crashes caused by outdated MCO dependency. This improvement enables Compliance Operator to support Ignition 3.4, which solves the issue. link:https://issues.redhat.com/browse/OCPBUGS-18025[OCPBUGS-18025]
```
`api-checks-pod` should be in backticks.

Probably should spell out MCO as well. 

"Machine Config Operator (MCO)"

GroceryBoyJr replied:
Done, thank you!

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Previously, when Compliance Operator remediations without variables set were applied to the entire cluster, this could result in stuck nodes even though the remediation has been applied. link:https://issues.redhat.com/browse/OCPBUGS-3864[OCPBUGS-3864]
+
+* Previously, Compliance Operator remediations would be shown as "unapplied" after scans for compliance were completed. link:https://issues.redhat.com/browse/OCPBUGS-8041[OCPBUGS-8041]
```
`uanpplied` should be in backticks, this is state of the scan (output).

GroceryBoyJr replied:
Done, thank you!

sheriff-rh reviewed (commented):
Some comments regarding backticks, attributes, and spelling out MCO for the first time we use it. 

In addition, I would consider writing bug summaries as two sentences (minimum). Example: 

"Before this update <describe the issue>. With this update <describe the fix>.

sheriff-rh commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+[id="compliance-operator-1-4-0-new-features-and-enhancements"]
+=== New features and enhancements
+
+* Users can now pause scan schedules by setting the `ScanSetting.suspend` attribute to `True`. This allows users to suspend a scan schedule and reactivate it without the need to delete and re-create the `ScanSettingBinding`. This simplifies pausing scan schedules during maintenance periods. link:https://issues.redhat.com/browse/CMP-2123[CMP-2123]
```
One final nit with the PR @GroceryBoyJr  - be sure to put all links in the enhancements and bugs in **BOLD**, and the entire `link:` statement goes in parenthesis: 

`(link:https://issues.redhat.com/browse/CMP-2123[*CMP-2123*])`

See more examples in CO 1.1.0, around line 200 or so on this file.

GroceryBoyJr replied:
@sheriff-rh This is done, thank you!

xiaojiey commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Previously, `rhcos4-audit-rules-login-events-faillock` would fail even after auto-remediation has been applied. With this update, `rhcos4-audit-rules-login-events-faillock` failure locks are now applied correctly after auto-remediation. (link:https://issues.redhat.com/browse/OCPBUGS-24594[*OCPBUGS-24594*])
+
+* Previously, Compliance Operator would fail when used with Ignition 3.4. This update enables Compliance Operator to support Ignition 3.4, which solves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-18025[*OCPBUGS-18025*])
```
This should already been released with last release COv1.3.1

xiaojiey commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Previously, Compliance Operator would fail when used with Ignition 3.4. This update enables Compliance Operator to support Ignition 3.4, which solves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-18025[*OCPBUGS-18025*])
+
+* Previously, when Compliance Operator remediations without variables set were applied to the entire cluster, this could result in stuck nodes even though the remediation has been applied. With this update, remediations without variables set, when applied to the entire cluster, will not result in stuck nodes. (link:https://issues.redhat.com/browse/OCPBUGS-3864[*OCPBUGS-3864*])
```
This should already been released long time ago, in Feb of 2023?

GroceryBoyJr replied:
@xiaojiey Correct, and THANK YOU! I have removed OCPBUGS-3864 from the release note and I see it has been removed from the ERRATA.

xiaojiey commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* With this update, `rprivate` default mount propagation is now handled correctly for root volume mounts of pods that rely on multipathing. (link:https://issues.redhat.com/browse/OCPBUGS-17494[*OCPBUGS-17494*])
+
+* Previously, `rhcos4-audit-rules-login-events-faillock` would fail even after auto-remediation has been applied. With this update, `rhcos4-audit-rules-login-events-faillock` failure locks are now applied correctly after auto-remediation. (link:https://issues.redhat.com/browse/OCPBUGS-24594[*OCPBUGS-24594*])
```
Maybe better add rule before the rule name?
`Previously, rule `rhcos4-audit-rules-login-events-faillock` ...`

GroceryBoyJr replied:
@xiaojiey I have made this change. Thank you!

xiaojiey commented:
/hold
with COv1.4.0, there is important change for https://docs.openshift.com/container-platform/4.14/security/compliance_operator/co-scans/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation, the Procedure 1 is not needed any more as you don't need a tailoredprofile created before scanning the custom node pool.

GroceryBoyJr commented:
@xiaojiey THANK YOU for that feedback. I have updated the section to remove the first step in the procedure, since tailoredprofile does not need to be created before scanning the custom node pool. Here is the new preview link for you to see the change: https://file.rdu.redhat.com/jbrigman/compliance-operator-140/security/compliance_operator/co-scans/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation

xiaojiey commented:
For the procedure 2 in https://file.rdu.redhat.com/jbrigman/compliance-operator-140/security/compliance_operator/co-scans/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation, 
The below yaml is not needed any more:
```
- apiGroup: compliance.openshift.io/v1alpha1
  kind: TailoredProfile
  name: cis-example-tp
```
And you can remove below descriptions as it is not needed any more:
`The Compliance Operator checks the runtime KubeletConfig through the Node/Proxy API object and then uses variables such as ocp-var-role-master and ocp-var-role-worker to determine the nodes it performs the check against. In the ComplianceCheckResult, the KubeletConfig rules are shown as ocp4-cis-kubelet-*. The scan passes only if all selected nodes pass this check.`

GroceryBoyJr commented:
@xiaojiey The requested change has been completed, PTAL. Thank  You!

GroceryBoyJr commented:
> For the procedure 2 in https://file.rdu.redhat.com/jbrigman/compliance-operator-140/security/compliance_operator/co-scans/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation, The below yaml is not needed any more:
> 
> ```
> - apiGroup: compliance.openshift.io/v1alpha1
>   kind: TailoredProfile
>   name: cis-example-tp
> ```
> 
> And you can remove below description as it is not needed any more: `The Compliance Operator checks the runtime KubeletConfig through the Node/Proxy API object and then uses variables such as ocp-var-role-master and ocp-var-role-worker to determine the nodes it performs the check against. In the ComplianceCheckResult, the KubeletConfig rules are shown as ocp4-cis-kubelet-*. The scan passes only if all selected nodes pass this check.`

This is completed, thank you @xiaojiey !

xiaojiey commented on modules/compliance-custom-node-pools.adoc:
```diff
@@ -75,14 +54,8 @@ profiles:
 - apiGroup: compliance.openshift.io/v1alpha1
   kind: TailoredProfile
   name: cis-example-tp
-settingsRef:
```
The settingsRef should not be deleted. You need to remove the TailoredProfile before this line. 
```
- apiGroup: compliance.openshift.io/v1alpha1
  kind: TailoredProfile
  name: cis-example-tp
```

GroceryBoyJr replied:
This is done, thank you @xiaojiey !

xiaojiey replied:
Still not correct. The ScanSettingBinding CR should be as below. Thanks.
```
apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: cis
  namespace: openshift-compliance
profiles:
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-cis
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-cis-node
settingsRef:
  apiGroup: compliance.openshift.io/v1alpha1
  kind: ScanSetting
  name: default
```

GroceryBoyJr replied:
@xiaojiey Done, thank you!

xiaojiey replied:
Need to remove
```
- apiGroup: compliance.openshift.io/v1alpha1
  name: cis-example-tp
```

GroceryBoyJr replied:
@xiaojiey This is done, thank you!

GroceryBoyJr replied:
@xiaojiey I believe I have the YAML correct now, PTAL.

xiaojiey commented:
Sorry, this line also need to be deleted:
`"If your cluster uses custom node pools outside the default `worker` and `master` node pools, you must supply additional variables to ensure the Compliance Operator aggregates a configuration file for that node pool."`

@GroceryBoyJr @rhmdnd Can we highlight in the release notes that "If your cluster uses custom node pools outside the default `worker` and `master` node pools, you don't need to  supply additional variables any more to ensure the Compliance Operator aggregates a configuration file for that node pool."? Besides, I believe there was an article about this. Do we need to add link to it? Thanks.

GroceryBoyJr commented:
> Sorry, this line also need to be deleted: `"If your cluster uses custom node pools outside the default `worker`and`master` node pools, you must supply additional variables to ensure the Compliance Operator aggregates a configuration file for that node pool."`
> 
> @GroceryBoyJr @rhmdnd Can we highlight in the release notes that "If your cluster uses custom node pools outside the default `worker` and `master` node pools, you don't need to supply additional variables any more to ensure the Compliance Operator aggregates a configuration file for that node pool."? Besides, I believe there was an article about this. Do we need to add link to it? Thanks.

@xiaojiey OK! Working now, please stand by.

GroceryBoyJr commented:
@xiaojiey I've removed the line from the docs and added your requested information to "New features and enhancements". I am looking at the errata to see what issue applies to the new feature you indicated.

GroceryBoyJr commented:
@xiaojiey I believe I am caught up with the edits you have requested. (I am still looking for a link in the errata, however.) Thank you so much.

GroceryBoyJr commented:
@xiaojiey Here is where I'm at with the YAML, but it does no tlook right to me, apiGroup: is repeated under profiles:

apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: cis
  namespace: openshift-compliance
profiles:
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-cis
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-cis-node
- settingsRef:
  apiGroup: compliance.openshift.io/v1alpha1
  kind: ScanSetting
  name: default

GroceryBoyJr commented:
@xiaojiey Stand by, updating.

GroceryBoyJr commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Compliance Operator now supports profile names in `ComplianceRules`. (link:https://issues.redhat.com/browse/CMP-2126[*CMP-2126*])
+
+* Compliance Operator compatibility with improved `cronjob` API improvements is available in this release. (link:https://issues.redhat.com/browse/CMP-2310[*CMP-2310*])
```
I'm glad to keep or remove. It's there because it's a public issue you have resolved.

rhmdnd commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* With this update, `rprivate` default mount propagation is now handled correctly for root volume mounts of pods that rely on multipathing. (link:https://issues.redhat.com/browse/OCPBUGS-17494[*OCPBUGS-17494*])
+
+* Previously, Compliance Operator remediations would show as `unapplied` after scans for compliance completed. With this release, remediations are now correctly shown after compliance scans complete. (link:https://issues.redhat.com/browse/OCPBUGS-8041[*OCPBUGS-8041*])
```
The fix for this bug was specific to a single rule and remediation, which we could describe here:

_Previously, the Compliance Operator would generate a remediation for `coreos_vsyscall_kernel_argument`, despite having a hardcoded `INFO` status, making it impossible to reconcile the rule even while applying the remediation. With release 1.4.0, the `coreos_vsyscall_kernel_argument` rule properly  evaluates kernel arguments and generates an appropriate remediation._

GroceryBoyJr replied:
@rhmdnd May I slightly simplify the background of the problem in order to focus on a "problem/fix" note model? For example: 

Previously, the Compliance Operator would generate a remediation for coreos_vsyscall_kernel_argument without reconciling the rule even while applying the remediation. With release 1.4.0, the coreos_vsyscall_kernel_argument rule properly evaluates kernel arguments and generates an appropriate remediation.

rhmdnd commented on security/compliance_operator/compliance-operator-release-notes.adoc:
```diff
@@ -15,6 +15,37 @@ For an overview of the Compliance Operator, see xref:../../security/compliance_o
+
+* Before this update, rule `rhcos4-audit-rules-login-events-faillock` would fail even after auto-remediation has been applied. With this update, `rhcos4-audit-rules-login-events-faillock` failure locks are now applied correctly after auto-remediation. (link:https://issues.redhat.com/browse/OCPBUGS-24594[*OCPBUGS-24594*])
+
+* Previously, upgrades from Compliance Operator 1.3.1 to Compliance Operator 1.4.0 would cause OVS rules scan results to go from `PASS` to `NOT-APPLICABLE`. With this update, OVS rules scan results now show `PASS` (link:https://issues.redhat.com/browse/OCPBUGS-25323[*OCPBUGS-25323*])
```
This one is tricky since we caught it before the release, and before users could upgrade from 1.3.1 to 1.4.0. Users shouldn't seen any issues with OVS rules, on 1.4.0 compared to 1.3.1.

GroceryBoyJr replied:
@rhmdnd Oh dang, OK. Leave it as is, or delete?

sheriff-rh commented:
I am removing the hold because it appears the requested changes have been implemented. If further editing is required, open an OCPBUG and assign it to @GroceryBoyJr . 

Merging this PR because the errata is now live.

sheriff-rh commented:
/cherrypick enterprise-4.15

sheriff-rh commented:
/cherrypick enterprise-4.13

sheriff-rh commented:
/cherrypick enterprise-4.12

sheriff-rh commented:
/cherrypick enterprise-4.11
